src/Chapter2.hs

```diff
@@ -348,8 +365,8 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = take ((length l) `div` 2) l
```

Comment:
Great! 💯 
Note that you can omit the brackets here as you are using the infix form of the `div` function:

```suggestion
firstHalf l = take (length l `div` 2) l
```